### PR TITLE
Fixed lp:1633957 "sql/field.h:918: bool Field::has_associated_compres…

### PR DESCRIPTION
**Pull request by percona-ysorokin**
…sion_dictionary() const: Assertion `zip_dict_name.str == 0 || column_format() == COLUMN_FORMAT_TYPE_COMPRESSED' failed"

Backported from 5.7.

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Not GCA as it does not include "compressed columns" yet.

**Comment by percona-ysorokin** [comment]
Jenkins build link
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1498/

File: mysql-test/suite/innodb/r/xtradb_compressed_columns_with_dictionaries.result [hunk 1]
```diff
@@ -178,8 +178,8 @@ original_dictionary_references_cleaned
 modified_dictionary_references_cleaned
 1
 
-# check if "CREATE TABLE ... AS SELECT * FROM ..." does not lose "COLUMN_FORMAT COMPRESSED" attribute (implicit SE)
-*** "a BLOB COLUMN_FORMAT COMPRESSED" -> "a BLOB COLUMN_FORMAT COMPRESSED", ALGOTIRHM=DEFAULT
+# check if "CREATE TABLE ... AS SELECT * FROM ..." discards "COLUMN_FORMAT COMPRESSED" attribute (implicit SE)
+*** "a BLOB COLUMN_FORMAT COMPRESSED" -> "a BLOB", ALGOTIRHM=DEFAULT
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/r/xtradb_compressed_columns_with_dictionaries.result]: "ALGORITHM"

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by percona-ysorokin** [comment]
Updated Jenkins build link
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1500/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**